homeassistant/components/ps4/media_player.py

```diff
@@ -270,20 +271,41 @@ def add_games(self, title_id, app_name):
 
     def get_device_info(self, status):
         """Set device info for registry."""
-        _sw_version = status['system-version']
-        _sw_version = _sw_version[1:4]
-        sw_version = "{}.{}".format(_sw_version[0], _sw_version[1:])
-        self._info = {
-            'name': status['host-name'],
-            'model': 'PlayStation 4',
-            'identifiers': {
-                (PS4_DOMAIN, status['host-id'])
-            },
-            'manufacturer': 'Sony Interactive Entertainment Inc.',
-            'sw_version': sw_version
-        }
-
-        self._unique_id = format_unique_id(self._creds, status['host-id'])
+        # If cannot get status on startup, assume info from registry.
+        if status is None:
+            e_registry = self.hass.data['entity_registry']
```

Comment:
Use `async_get_registry` in helpers.entity_registry. It's a coroutine, so we should think about how to refactor this. Perhaps implement `async_update` and change `update` to `_update` and schedule that on the executor thread pool from `async_update`. Then we have options in `async_update` to await coroutines as needed and can pass on info from there.

Reply:
Ok Ill see if I can implement an async versions of the component and dependency first.